syntax = "proto3";

package student.academic.v1;

message Transparency {
    string register_number = 1;
    uint32 entry_id = 2;
    uint32 employee_min = 3;
    uint32 employee_max = 4;
    bool refuse_expenses = 5;
    optional uint64 expenses_min = 6;
    optional uint64 expenses_max = 7;
    optional string fiscal_year = 8;
    bool refuse_allowance = 9;
    bool refuse_donation = 10;
    optional bool donation_information_required = 11;
    string firstPublicationDate = 12;
    bool account_inactive = 13;
    repeated LobbyistIdentity lobbyist = 14;
    repeated LegalRepresentatives persons = 15;
    repeated NamedEmployees employees = 16;
    repeated MembershipEntries organisations = 17;
    repeated Donators companies = 18;
    repeated fieldsOfInterest fields = 19;
    repeated ClientOrganizations clients = 20;
    string name = 21;
}

message LobbyistIdentity {
    string identity = 1;
    string name = 2;
    Address adress = 3;
}

message Address {
    string street = 1;
    uint32 streetNumber = 2;
    uint32 zipCode = 3;
    string city = 4;
    string country = 5;

}

message LegalRepresentatives {
    string first_name = 1;
    string last_name = 2;
    optional string function = 3;
}

message NamedEmployees {
    string first_name = 1;
    string last_name = 2;
}

message MembershipEntries {
    string name = 1;
}

message Donators {
    string name = 1;
    string fiscal_year = 2;
    uint64 donation_euro = 3;
}

message fieldsOfInterest{
    string name = 1;
    string description = 2;
}

message ClientOrganizations{
    string name = 1;
    Address adress = 2;
}